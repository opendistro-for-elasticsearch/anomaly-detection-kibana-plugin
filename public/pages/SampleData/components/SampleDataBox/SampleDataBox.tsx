/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import ContentPanel from '../../../../components/ContentPanel/ContentPanel';
import {
  EuiButton,
  EuiFlexGroup,
  EuiFlexItem,
  EuiTitle,
  EuiLink,
} from '@elastic/eui';
import { PLUGIN_NAME } from '../../../../utils/constants';

interface SampleDataBoxProps {
  title: string;
  icon: any;
  description: string;
  loadDataButtonDescription: string;
  onOpenFlyout(): void;
  onLoadData(): void;
  isLoadingData: boolean;
  isDataLoaded: boolean;
  detectorId: string;
}

export const SampleDataBox = (props: SampleDataBoxProps) => {
  return (
    <div style={{ height: 'auto' }}>
      <ContentPanel
        title={
          <div
            style={{
              display: 'flex',
              flexDirection: 'row',
              alignContent: 'flexStart',
            }}
          >
            {props.icon}
            <EuiTitle size="s">
              <h2 style={{ marginLeft: '12px', marginTop: '-3px' }}>
                {props.title}
              </h2>
            </EuiTitle>
            <EuiLink
              style={{ marginLeft: '12px' }}
              onClick={props.onOpenFlyout}
            >
              Info
            </EuiLink>
          </div>
        }
        titleSize="s"
        badgeLabel={props.isDataLoaded ? 'INSTALLED' : undefined}
      >
        <EuiFlexGroup direction="column" gutterSize="l">
          <EuiFlexItem grow={false} style={{ height: '70px' }}>
            <p
              style={{
                textAlign: 'left',
                lineHeight: 1.4,
                maxHeight: 4.2,
              }}
            >
              {props.description}
            </p>
          </EuiFlexItem>
          <EuiFlexGroup
            style={{
              height: '100px',
              marginTop: '0px',
              marginBottom: '0px',
            }}
            direction="column"
            alignItems="center"
          >
            <EuiFlexItem grow={false}>
              <EuiButton
                style={{ width: '300px' }}
                data-test-subj="loadDataButton"
                disabled={props.isLoadingData || props.isDataLoaded}
                isLoading={props.isLoadingData}
                onClick={() => {
                  props.onLoadData();
                }}
              >
                {props.isLoadingData
                  ? 'Creating detector'
                  : props.isDataLoaded
                  ? 'Detector created'
                  : props.loadDataButtonDescription}
              </EuiButton>
            </EuiFlexItem>
            <EuiFlexItem grow={false}>
              {props.isDataLoaded ? (
                <EuiLink href={`${PLUGIN_NAME}#/detectors/${props.detectorId}`}>
                  View detector and sample data
                </EuiLink>
              ) : null}
            </EuiFlexItem>
          </EuiFlexGroup>
        </EuiFlexGroup>
      </ContentPanel>
    </div>
  );
};
