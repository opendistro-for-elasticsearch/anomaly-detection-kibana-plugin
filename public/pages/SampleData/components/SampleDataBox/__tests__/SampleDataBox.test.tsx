/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { render } from '@testing-library/react';
import { EuiIcon } from '@elastic/eui';
import { SampleDataBox } from '../SampleDataBox';

const defaultProps = {
  title: 'Sample title',
  icon: <EuiIcon type="alert" />,
  description: 'Sample description',
  loadDataButtonDescription: 'Sample button description',
  onLoadData: jest.fn(),
  isLoadingData: false,
  isDataLoaded: false,
  detectorId: 'sample-detector-id',
};

describe('<SampleDataBox /> spec', () => {
  describe('Data not loaded', () => {
    test('renders component', () => {
      const { container, getByText } = render(
        <SampleDataBox {...defaultProps} />
      );
      expect(container.firstChild).toMatchSnapshot();
      getByText('Sample title');
      getByText('Sample description');
      getByText('Sample button description');
    });
  });
  describe('Data is loading', () => {
    test('renders component', () => {
      const { container, getByText } = render(
        <SampleDataBox {...defaultProps} isLoadingData={true} />
      );
      expect(container.firstChild).toMatchSnapshot();
      getByText('Sample title');
      getByText('Sample description');
      getByText('Creating detector');
    });
  });
  describe('Data is loaded', () => {
    test('renders component', () => {
      const { container, getByText } = render(
        <SampleDataBox {...defaultProps} isDataLoaded={true} />
      );
      expect(container.firstChild).toMatchSnapshot();
      getByText('Sample title');
      getByText('Sample description');
      getByText('Detector created');
    });
  });
});
