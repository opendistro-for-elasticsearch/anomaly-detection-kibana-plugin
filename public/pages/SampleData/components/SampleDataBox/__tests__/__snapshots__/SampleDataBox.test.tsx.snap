// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SampleDataBox /> spec Data is loaded renders component 1`] = `
<div
  style="height: auto;"
>
  <div
    class="euiPanel euiPanel--paddingMedium euiPanel--hasBetaBadge"
    style="padding: 20px;"
  >
    <span
      class="euiPanel__betaBadgeWrapper"
    >
      <span
        class="euiBetaBadge euiPanel__betaBadge"
        title="INSTALLED"
      >
        INSTALLED
      </span>
    </span>
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px;"
    >
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          >
            <div
              style="display: flex;"
            >
              <svg
                aria-hidden="true"
                class="euiIcon euiIcon--medium euiIcon-isLoading"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              />
              <h2
                class="euiTitle euiTitle--small"
                style="margin-left: 12px; margin-top: -3px;"
              >
                Sample title
              </h2>
              <button
                class="euiLink euiLink--primary"
                style="margin-left: 12px;"
                type="button"
              >
                Info
              </button>
            </div>
          </div>
        </div>
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          />
        </div>
      </div>
      <div
        class="euiFlexItem euiFlexItem--flexGrowZero"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          />
        </div>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
    />
    <div
      style="padding: 10px 0px;"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionColumn euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
          style="height: 70px;"
        >
          <p
            style="text-align: left; line-height: 1.4; max-height: 4.2px;"
          >
            Sample description
          </p>
        </div>
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--directionColumn euiFlexGroup--responsive"
          style="height: 100px; margin-top: 0px; margin-bottom: 0px;"
        >
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <button
              class="euiButton euiButton--primary"
              data-test-subj="loadDataButton"
              disabled=""
              style="width: 300px;"
              type="button"
            >
              <span
                class="euiButton__content"
              >
                <span
                  class="euiButton__text"
                >
                  Detector created
                </span>
              </span>
            </button>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <a
              class="euiLink euiLink--primary"
              href="opendistro-anomaly-detection-kibana#/detectors/sample-detector-id"
              rel="noreferrer"
            >
              View detector and sample data
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<SampleDataBox /> spec Data is loading renders component 1`] = `
<div
  style="height: auto;"
>
  <div
    class="euiPanel euiPanel--paddingMedium"
    style="padding: 20px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px;"
    >
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          >
            <div
              style="display: flex;"
            >
              <svg
                aria-hidden="true"
                class="euiIcon euiIcon--medium euiIcon-isLoading"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              />
              <h2
                class="euiTitle euiTitle--small"
                style="margin-left: 12px; margin-top: -3px;"
              >
                Sample title
              </h2>
              <button
                class="euiLink euiLink--primary"
                style="margin-left: 12px;"
                type="button"
              >
                Info
              </button>
            </div>
          </div>
        </div>
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          />
        </div>
      </div>
      <div
        class="euiFlexItem euiFlexItem--flexGrowZero"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          />
        </div>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
    />
    <div
      style="padding: 10px 0px;"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionColumn euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
          style="height: 70px;"
        >
          <p
            style="text-align: left; line-height: 1.4; max-height: 4.2px;"
          >
            Sample description
          </p>
        </div>
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--directionColumn euiFlexGroup--responsive"
          style="height: 100px; margin-top: 0px; margin-bottom: 0px;"
        >
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <button
              class="euiButton euiButton--primary"
              data-test-subj="loadDataButton"
              disabled=""
              style="width: 300px;"
              type="button"
            >
              <span
                class="euiButton__content"
              >
                <span
                  class="euiLoadingSpinner euiLoadingSpinner--medium euiButton__spinner"
                />
                <span
                  class="euiButton__text"
                >
                  Creating detector
                </span>
              </span>
            </button>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<SampleDataBox /> spec Data not loaded renders component 1`] = `
<div
  style="height: auto;"
>
  <div
    class="euiPanel euiPanel--paddingMedium"
    style="padding: 20px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px;"
    >
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          >
            <div
              style="display: flex;"
            >
              <svg
                aria-hidden="true"
                class="euiIcon euiIcon--medium euiIcon-isLoading"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              />
              <h2
                class="euiTitle euiTitle--small"
                style="margin-left: 12px; margin-top: -3px;"
              >
                Sample title
              </h2>
              <button
                class="euiLink euiLink--primary"
                style="margin-left: 12px;"
                type="button"
              >
                Info
              </button>
            </div>
          </div>
        </div>
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          />
        </div>
      </div>
      <div
        class="euiFlexItem euiFlexItem--flexGrowZero"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          />
        </div>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
    />
    <div
      style="padding: 10px 0px;"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionColumn euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
          style="height: 70px;"
        >
          <p
            style="text-align: left; line-height: 1.4; max-height: 4.2px;"
          >
            Sample description
          </p>
        </div>
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--directionColumn euiFlexGroup--responsive"
          style="height: 100px; margin-top: 0px; margin-bottom: 0px;"
        >
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <button
              class="euiButton euiButton--primary"
              data-test-subj="loadDataButton"
              style="width: 300px;"
              type="button"
            >
              <span
                class="euiButton__content"
              >
                <span
                  class="euiButton__text"
                >
                  Sample button description
                </span>
              </span>
            </button>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
